### PR TITLE
feat: uploaded files

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |Y|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

Allow to have a view on uploaded files

### CODE REVIEW

File: src/Controller/ContentManagement/FileController.php [hunk 1]
```diff
@@ -41,6 +41,26 @@ public function downloadFileAction(string $sha1, Request $request, FileService $
...
 
+    /**
+     * @Route("/admin/file/{id}/soft-delete" , name="ems_file_soft_delete", methods={"POST","HEAD"})
+     */
+    public function softDeleteFileAction(string $id, FileService $fileService): Response
+    {
+        $fileService->removeSingleFileEntity([$id]);
+
+        return $this->redirectToRoute('ems_core_uploaded_file_index');
+    }
```
Comment (dameert) [inline src/Controller/ContentManagement/FileController.php:52]: +1 for this really slim controller!

File: src/Controller/UploadedFileController.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            ];
+        });
+        $column->setRouteTarget('_blank');
+        $table->addColumn('uploaded-file.index.column.sha1', 'sha1');
+        $table->addColumn('uploaded-file.index.column.type', 'type');
+        $table->addColumn('uploaded-file.index.column.username', 'user');
+        $tableColumn = $table->addColumn('uploaded-file.index.column.created', 'created');
+        $tableColumn->setDateTimeProperty(true);
+        $tableColumn = $table->addColumn('uploaded-file.index.column.size', 'size');
+        $tableColumn->setFormatBytes(true);
```
Comment (dameert) [inline src/Controller/UploadedFileController.php:51]:
```suggestion
        $table
            ->addColumn('uploaded-file.index.column.name', 'name')
            ->setRoutePath('ems_file_download', function (UploadedAsset $data) {
                return [
                    'sha1' => $data->getSha1(),
                    'type' => $data->getType(),
                    'name' => $data->getName(),
                ];
            })
            ->setRouteTarget('_blank');
        $table->addColumn('uploaded-file.index.column.sha1', 'sha1');
        $table->addColumn('uploaded-file.index.column.type', 'type');
        $table->addColumn('uploaded-file.index.column.username', 'user');
        $table
            ->addColumn('uploaded-file.index.column.created', 'created')
            ->setDateTimeProperty(true);
        $table
            ->addColumn('uploaded-file.index.column.size', 'size')
            ->setFormatBytes(true);
```

Comment (dameert) [reply src/Controller/UploadedFileController.php:51]: let me know what you think about this strategy @Davidmattei and @theus77

Comment (theus77) [reply src/Controller/UploadedFileController.php:51]: I have stated another PR in order to simplify this. For instance with Column classes by type of column such as UserTableColumn, DatetimeTableColumn, BytesTableColumn

Comment (Davidmattei) [reply src/Controller/UploadedFileController.php:51]: Perfect!

**Review by Davidmattei — APPROVED** [review]

**Pull request merged**